chapters/resampling.qmd

```diff
@@ -584,6 +586,144 @@ Should we use the bootstrap to compare models? Its small variance is enticing, b
+#| echo: false
+#| fig-cap: An example of one split produced by block cross-validation on the forestation data. The analysis (magenta) and assessment (purple) sets are shown with locations in the buffer in black. The empty spaces represent the data reserved for the test set (see @fig-forested-split). 
+
+leaflet() %>%
+  addProviderTiles(providers$CartoDB.PositronNoLabels) %>%
+  addCircles(
+    data = forested_cv_split_info,
+    lng = ~ longitude,
+    lat = ~ latitude,
+    color = ~ group_col,
+    fillColor = ~ group_col,
+    fill = TRUE,
+    opacity = .01,
+    fillOpacity = 3 / 4,
+    radius = 1500,
+    popup = htmltools::htmlEscape(forested_cv_split_info$group)
+  ) %>%
+  setView(lng = -120.3945, lat = 47.33047, zoom = 7)
+```
+
+Our training set contains `r format(nrow(forested_train), big.mark = ",")` locations. Across each of the ten folds, every location appears once in the analysis and assessment sets. 
+
+Just as with standard _V_-fold cross-validation, the number of folds can be chosen based on bias and variance concerns. Unlike standard resampling, any investigation into this topic would depend on a specific spatial autocorrelation structure. As such, we have decided to follow the recommendations for non-spatial data and use _V_=10. If we find that the variation in our statistics estimated from ten resamples is too large, we can employ the same process of repeating the block resampling process using different random number seeds. For our data, the average number of locations in the assessment sets is `r floor(mean(map_int(forested_rs$splits, ~ length(.x$out_id))))` and this appears large enough to produce sufficiently precise performance statistics. 
+
+## Grouped or Multi-Level Data {#sec-multilevel-resampling}
+
+When rows of data are correlated for other reasons, such as those discussed in @sec-multilevel-splitting, we can extend the techniques described above. Consider the previously mentioned example of having the purchase histories of many users from a retail database. If we had a training set of 10,000 instances that included 100 customers, we could conduct _V_-fold cross-validation to determine which _customers_ would go into the analysis and assessment sets. As with the initial split, any rows associated with specific customers would be placed into the same partition (i.e., analysis or assessment). 
+
+## Frequently Asked Questions {#sec-resampling-faq}
+
+These questions represent areas of confusion regarding resampling. 
+
+#### Is it bad to get different results from each resample?   {#sec-resampling-faq-diff-results .unnumbered}
+
+The point of resampling is to see how the model pipeline changes when the data changes. There is a good chance that some important estimates differ from resample to resample. For example, when algorithmically filtering predictors, you might get _B_ different predictor lists, and that’s okay. It does give you a sense of how much noise is in the part of your model pipeline.
```

Comment:
```suggestion
The point of resampling is to see how the model pipeline changes when the data changes. There is a good chance that some important estimates differ from resample to resample. For example, when algorithmically filtering predictors, you might get _B_ different predictor lists, and that’s okay. It does give you a sense of how much noise is in the part of your model pipeline.

It can be good to examine the distribution of resampling performance.  Occasionally, we may find a resample that has unusual performance with respect to the rest of the resamples.  This may merit further investigation as to why the resample is unusual.
```